feat(maxMarginalRelevanceSearch): add maxMarginalRelevanceSearch methods

Itzaprado opened the pull request.

Itzaprado committed:
feat(maxMarginalRelevanceSearch): add maxMarginalRelevanceSearch methods

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/vectorStore.ts:
```diff
@@ -327,6 +329,83 @@ class PostgresVectorStore extends VectorStore {
+
+  async maxMarginalRelevanceSearch(
+    query: string,
+    options?: MaxMarginalRelevanceSearchOptions<this["FilterType"]>,
+    _callbacks?: Callbacks | undefined
```
Interface says this isn't optional. We will also ignore callbacks for now
```suggestion
    options: MaxMarginalRelevanceSearchOptions<this["FilterType"]>
```

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/vectorStore.ts:
```diff
@@ -327,6 +329,83 @@ class PostgresVectorStore extends VectorStore {
+    const vector = await this.embeddings.embedQuery(query);
+    return await this.maxMarginalRelevantSearchByVector(
+      vector,
+      options?.k,
+      options?.fetchK,
+      options?.lambda,
+      options?.filter
```
To simplify this code, you could just pass through the options rather than each parameter

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/vectorStore.ts:
```diff
@@ -327,6 +329,83 @@ class PostgresVectorStore extends VectorStore {
+    return docsList;
+  }
+
+  async maxMarginalRelevanceSearchWithScoreByVector(vector: number[], k?: number, fetchK?: number, lambdaMult?: number, filter?: this["FilterType"]): Promise<Document[]> {
```
This function and maxMarginalRelevantSearchByVector aren't defined by the JS interface so let's delete them for now and move this logic to "maxMarginalRelevanceSearch"

Itzaprado committed:
refactor: update maxMarginalRelevanceSearch and tests
[ ] move all logic to maxMarginalRelevanceSearch
[ ] remove unnecessary tests
[ ] update test to use logic updated

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/vectorStore.ts:
```diff
@@ -327,6 +328,49 @@ class PostgresVectorStore extends VectorStore {
+
+  async maxMarginalRelevanceSearch(
+    query: string,
+    options?: MaxMarginalRelevanceSearchOptions<this["FilterType"]>
```
```suggestion
    options: MaxMarginalRelevanceSearchOptions<this["FilterType"]>
```

averikitsch committed:
Update libs/langchain-google-cloud-sql-pg/src/vectorStore.ts
